lib/ui/common/ActionSheetListItem.dart

```diff
@@ -0,0 +1,175 @@
+      {this.icon, this.checkBoxIcon});
+}
+
+class ActionSheetListItemStyle {
+  final Color actionItemBackgroundColor;
+
+  final TextStyle actionTextStyle;
+  final TextStyle destructiveTextStyle;
+
+  final EdgeInsets actionItemEdgePadding;
+
+  final double actionItemHeight;
+  final double iconLineSpace;
+  final double actionItemElevation;
+  final double iconHeight;
+
+  final int maxLines;
+
+  ActionSheetListItemStyle(
+      {this.actionItemBackgroundColor,
+      this.actionTextStyle,
+      this.destructiveTextStyle,
+      this.actionItemEdgePadding,
+      this.actionItemHeight,
+      this.iconLineSpace,
+      this.actionItemElevation,
+      this.iconHeight,
+      this.maxLines});
+
+  factory ActionSheetListItemStyle.defaultStyle() {
+    return ActionSheetListItemStyle(
+      actionItemBackgroundColor: const Color(0x00000000),
+      actionTextStyle: const TextStyle(
+          fontSize: 17,
+          fontWeight: FontWeight.normal,
+          color: Color(0xFF1a1b46)),
+      destructiveTextStyle: const TextStyle(
+          fontSize: 17,
+          fontWeight: FontWeight.normal,
+          color: Color(0xFFff6060)),
+      actionItemEdgePadding: const EdgeInsets.symmetric(horizontal: 32),
+      actionItemHeight: 70,
+      iconLineSpace: 21.5,
+      actionItemElevation: 0,
+      iconHeight: 20,
+      maxLines: 1,
+    );
+  }
+
+  factory ActionSheetListItemStyle.selectableStyle() {
+    return ActionSheetListItemStyle.defaultStyle().copyWith(iconHeight: 24);
+  }
+
+  ActionSheetListItemStyle copyWith(
+      {Color actionItemBackgroundColor,
+      TextStyle actionTextStyle,
+      TextStyle destructiveTextStyle,
+      EdgeInsets actionItemEdgePadding,
+      double actionItemHeight,
+      double iconLineSpace,
+      double actionItemElevation,
+      double iconHeight,
+      int maxLines}) {
+    return ActionSheetListItemStyle(
+        actionItemBackgroundColor:
+            actionItemBackgroundColor ?? this.actionItemBackgroundColor,
+        actionTextStyle: actionTextStyle ?? this.actionTextStyle,
+        destructiveTextStyle: destructiveTextStyle ?? this.destructiveTextStyle,
+        actionItemEdgePadding:
+            actionItemEdgePadding ?? this.actionItemEdgePadding,
+        actionItemHeight: actionItemHeight ?? this.actionItemHeight,
+        iconLineSpace: iconLineSpace ?? this.iconLineSpace,
+        actionItemElevation: actionItemElevation ?? this.actionItemElevation,
+        iconHeight: iconHeight ?? this.iconHeight,
+        maxLines: maxLines ?? this.maxLines);
+  }
+}
+
+class ActionSheetListItem extends StatelessWidget {
+  final ActionSheetListItemViewModel _viewModel;
+  final ActionSheetListItemStyle _style;
+  final int _numberOfItems;
+  final int _currentItem;
+
+  const ActionSheetListItem(
+      {Key key,
+      ActionSheetListItemViewModel viewModel,
+      ActionSheetListItemStyle style,
+      int numberOfActions,
```

Comment:
don't get this? this is a list item, why does it need to know number of actions and current action?

Reply:
It was just a temporal solution to build a divider between components, except in the last cell

Reply:
Now it's deleted